clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -404,14 +404,32 @@ boolean joinGroupIfNeeded(final Timer timer) {
             }
 
             if (future.succeeded()) {
-                // Duplicate the buffer in case `onJoinComplete` does not complete and needs to be retried.
-                ByteBuffer memberAssignment = future.value().duplicate();
-                onJoinComplete(generation.generationId, generation.memberId, generation.protocol, memberAssignment);
+                Generation gen;
```

Comment:
Nit: rename `gen` to `generationSnapshot`?

Reply:
renamed